src/UndockedRegFreeWinRT/UndockedRegFreeWinRT/catalog.cpp

```diff
@@ -81,72 +82,373 @@ struct component
 };
 
 static unordered_map<wstring, shared_ptr<component>> g_types;
+static unordered_set<wstring> g_assemblies;
 
-HRESULT WinRTLoadComponent(PCWSTR manifest_path)
+HRESULT LoadFromSxSManifest(std::wstring path)
+{
+    return WinRTLoadComponentFromFilePath(path.c_str());
+}
+
+HRESULT LoadFromEmbeddedManifest(std::wstring path)
+{
+    int resource = 0;
+    if (path.size() < 4)
+    {
+        return COR_E_ARGUMENT;
+    }
+    std::wstring ext = path.substr(path.size() - 4, path.size());
+    if (ext.compare(L".exe") == 0)
+    {
+        resource = 1;
+    }
+    else if (ext.compare(L".dll") == 0)
+    {
+        // I notice in testing, the original sxs allows this number to be also 1. 
+        resource = 2;
+    }
+    else
+    {
+        return COR_E_ARGUMENT;
+    }
+
+    HMODULE handle = LoadLibraryExW(path.c_str(), nullptr, LOAD_LIBRARY_AS_DATAFILE_EXCLUSIVE);
```

Comment:
Use WIL's `unique_hmodule` to not leak this HMODULE.

Reply:
Wasn't sure how to use unique_hmodule with LoadLibraryExW so I just use a wil::scope_exit